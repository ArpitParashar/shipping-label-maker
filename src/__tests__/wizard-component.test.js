import React from 'react';
import WizardComponent from '../core/components/wizard/wizard-component';
import ShallowRenderer from 'react-test-renderer/shallow';
import { shallow } from 'enzyme';
import Stepper from '@material-ui/core/Stepper';

let realUseContext;
let useContextMock;
// Setup mock
beforeEach(() => {
  realUseContext = React.useContext;
  useContextMock = React.useContext = jest.fn();
});
// Cleanup mock
afterEach(() => {
  React.useContext = realUseContext;
});

describe('Wizard component', () => {
  it('should render with props passed using context', () => {
    useContextMock.mockReturnValue([]);
    const steps = [
      {
        step: 1,
        component: 'xyz',
      },
      {
        step: 2,
        component: 'abc',
      },
    ];
    const onComplete = () => {};
    const element = new ShallowRenderer().render(
      <WizardComponent steps={steps} onComplete={onComplete} />
    );
    expect(element.props.children).toMatchSnapshot();
  });

  it('should change active step on action passed by the child step', () => {
    useContextMock.mockReturnValue([]);
    const steps = [
      {
        step: 1,
        component: 'xyz',
      },
      {
        step: 2,
        component: 'abc',
      },
    ];
    const onComplete = () => {};
    const element = shallow(<WizardComponent steps={steps} onComplete={onComplete} />);
    expect(element.find('#testNextStep')).toHaveLength(1);
    expect(element.find('#testPrevStep')).toHaveLength(1);
    const nextStep = element.find('#testNextStep');
    const prevStep = element.find('#testPrevStep');
    expect(element.find(Stepper)).toHaveLength(1);
    nextStep.simulate('click');
    expect(element.find(Stepper).props().activeStep).toBe(2);
    prevStep.simulate('click');
    expect(element.find(Stepper).props().activeStep).toBe(1);
  });
});
