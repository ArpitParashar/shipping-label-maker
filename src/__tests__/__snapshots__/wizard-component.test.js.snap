exports[`Wizard component should render with props passed using context 1`] = `
Array [
  <Unknown
    className="hide"
    id="testNextStep"
    onClick={[Function]} />,
  <Unknown
    className="hide"
    id="testPrevStep"
    onClick={[Function]} />,
  <h3
    className="red-text center text-darken-3">
     Welcome to the WIZARD!
</h3>,
  <div
    className="container">
    <Unknown
        activeStep={1}>
        <Unknown>
            <Unknown>
                1
            </Unknown>
        </Unknown>
        <Unknown>
            <Unknown>
                2
            </Unknown>
        </Unknown>
    </Unknown>
    <Unknown
        onAction={[Function]}
        wizardContext={undefined} />
</div>,
]
`;
